library/src/main/java/com/getstream/sdk/chat/view/MessageListView.java

```diff
@@ -129,24 +192,15 @@ private void parseAttr(Context context, @Nullable AttributeSet attrs) {
 
     // set the adapter and apply the style.
     @Override
-    public void setAdapter(Adapter adapter) {
-        throw new IllegalArgumentException("Use setAdapterWithStyle instead please");
-    }
-
-    private void setAdapterWithStyle(MessageListItemAdapter adapter) {
-        adapter.setStyle(style);
-        adapter.setGiphySendListener((Message message, GiphyAction action) -> onSendGiphyHandler.invoke(message, action));
-        setMessageClickListener(messageClickListener);
-        setMessageLongClickListener(messageLongClickListener);
-        setAttachmentClickListener(attachmentClickListener);
-        setReactionViewClickListener(reactionViewClickListener);
-        setUserClickListener(userClickListener);
-        setReadStateClickListener(readStateClickListener);
-        setMessageLongClickListener(messageLongClickListener);
+    @Deprecated
+    public final void setAdapter(Adapter adapter) {
+        throw new IllegalArgumentException("Use setMessageListItemAdapter instead please");
```

Comment:
The method being referenced here became `private` a while ago. Do we forbid setting a custom adapter on purpose, so that users only set custom things used by our adapter, such as the ViewHolder factory, the bubble helper, etc?

Reply:
@samiuelson, any idea why we changed this to `private`? It was your change a couple months ago. https://github.com/GetStream/stream-chat-android/commit/f7dfc64844e6350f2b17587a6a600c3563e8db7b#diff-778483d318fbb6393eb8add88b4fb9e1R127

Reply:
it just wasn't used outside the class, but probably it's a mistake on my end. feel free to make it public

Reply:
I think, previously the concept of working with `MessageListView` was to allow calling `setAdapterWithStyle` to set the adapter manually; or to call `init(Channel channel, User currentUser)` which calls `initAdapter()` internally. 

So, `setAdapterWithStyle` and `init(Channel channel, User currentUser)` were supposed to be public.

Reply:
Also, feel free to refactor this. Function names are a bit confusing.

Reply:
From out point of view, it's better if we can leave it `private`, as we then have control over what the adapter being used is... If we don't hear any complaints about this, I'd leave it as `private. It can be customized in a bunch of ways without swapping out our adapter implementation.